modelplugin/Junos-19.3R1.8/modelmain.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2019-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build modelplugin
+
+// A plugin for the YGOT model of Junos-19.3R1.8.
+package main
+
+import (
+	"fmt"
+	"github.com/onosproject/onos-config/modelplugin/Junos-19.3R1.8/junos_19_3R1_8"
+	"github.com/openconfig/gnmi/proto/gnmi"
+	"github.com/openconfig/goyang/pkg/yang"
+	"github.com/openconfig/ygot/ygot"
+)
+
+type modelplugin string
+
+const modeltype = "Junos"
+const modelversion = "19.3R1.8"
+const modulename = "junos.so.19.3R1.8"
```

Comment:
I'm not sure if a name like this is valid for a .so file - with a letter in it - I hope this won't break something down the line

Reply:
>I’m not sure if a name like this is valid for a .so file - with a letter in it - I hope this won’t break something down the line

@SeanCondon  I have come round to thinking we shouldn’t be using the (alphanumeric) YANG / device version as the .so version, and standard lib versioning approach mentioned at https://stackoverflow.com/questions/663209/can-someone-explain-about-linux-library-naming is needed.

Potentially we could need to track multiple plugins that match same version of the YANG  
e.g. if we find a bug in the generated go-code, and then need to fix code-generator tool then recreate the lib, we'll need a new version number for that .so file.

One simple way would be just incorporating the YANG version in the TYPE. e.g.
```
modelplugin/Junos-19.3R1.8.env:TYPE=Junos19_3R1_8
modelplugin/Junos-19.3R1.8.env:VERSION=1.0.0 
```
//Note: I’m deliberately avoiding dots and dashes in the TYPE to avoid compilation issues re. Package name.

 
This results in the  `modelplugin/Junos19_3R1_8-1.0.0/modelmain.go`
```
30:const modeltype = “Junos19_3R1_8"
31:const modelversion = “1.0.0”
32:const modulename = “junos19_3r1_8.so.1.0.0"
```

This results in these shared libs …
```
build/_output
├── devicesim.so.1.0.0
├── junos19_3r1_8.so.1.0.0
├── stratum.so.1.0.0
├── testdevice.so.1.0.0
└── testdevice.so.2.0.0
```

Later we can decide if the other four existing .so files need same update. WDYT ?

Reply:
I was thinking that the version number was a version of the **collection** of YANGs for a particular device. So I think junos.so.<yangs version> is fine, as long as the <yangs version> only has digits in it. So maybe junos.so.19.3.18 or junos.so.19.3.1.8 would work for me.

The way I had intended the versions to work was that the type would describe a family of versions. 

Re your comment about the bug in the generated go code, I think the versioning system of docker is there to take care of that scenario, as in future each plugin will be its own Docker image - so will be like 
`onosproject/junos_19.3R1_8:latest`

I know I was a bit undecided on the TST call last week, but since then I have figured out that I will create each plugin as a Docker image, and as many plugins as are wanted will be loaded alongside onos-config container in a "sidecar" fashion. This would mean that when I move to this pattern, the Helm chart would define and pull in the plugins needed, and they would be available in the same Pod because they are sidecars.